Sources/AppcuesKit/Presentation/UI/ExperienceStepViewController.swift

```diff
@@ -102,6 +110,42 @@ internal class ExperienceStepViewController: UIViewController {
             stepView.scrollView.scrollRectToVisible(frameInScrollView, animated: false)
         }
     }
+
+    private func decorateStickyContent(edge: ExperienceComponent.StackModel.StickyEdge, _ component: ExperienceComponent) {
+        // Must have the environmentObject so any actions in the sticky content can be applied.
+        let stickyContentVC = StickyHostingController(rootView: component.view.environmentObject(viewModel))
+
+        // Add the stick content to the parent controller.
+        addChild(stickyContentVC)
+        view.addSubview(stickyContentVC.view)
+        stickyContentVC.view.translatesAutoresizingMaskIntoConstraints = false
+
+        var constraints: [NSLayoutConstraint] = [
+            stickyContentVC.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            stickyContentVC.view.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+        ]
+
+        switch edge {
+        case .top:
+            constraints.append(stickyContentVC.view.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor))
+        case .bottom:
+            constraints.append(stickyContentVC.view.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor))
+        }
+
+        NSLayoutConstraint.activate(constraints)
+
+        stickyContentVC.didMove(toParent: self)
+
+        // Pass sticky content size changes to the parent controller to update the insets.
```

Comment:
are these insets also how you find the boundary for the mask for fading content behind the sticky area in that poc?

Reply:
Yeah, exactly. That defines a mask with a frame sized from the safe area